runtime/src/main/java/com/flipkart/flux/guice/interceptors/TransactionInterceptor.java

```diff
@@ -11,40 +11,45 @@
  * limitations under the License.
  */
 
-package com.flipkart.flux.util;
+package com.flipkart.flux.guice.interceptors;
 
 import org.aopalliance.intercept.MethodInterceptor;
 import org.aopalliance.intercept.MethodInvocation;
+import org.hibernate.HibernateException;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.context.internal.ManagedSessionContext;
+import javax.inject.Inject;
 
 /**
- * Provides transactional boundaries to methods which are annotated with {@link com.flipkart.flux.util.Transactional}.
+ * Provides transactional boundaries to methods which are annotated with {@link javax.transaction.Transactional}.
  * @author shyam.akirala
  */
 public class TransactionInterceptor implements MethodInterceptor {
 
-    private ThreadLocal<Session> threadLocalSession;
-
-    public TransactionInterceptor() {
-        threadLocalSession = new ThreadLocal<Session>();
-    }
+    @Inject
+    private SessionFactory sessionFactory;
 
     public Object invoke(MethodInvocation invocation) throws Throwable {
 
         Transaction transaction = null;
-        Session session = threadLocalSession.get();
+
+        Session session = null;
+        try {
+            session = sessionFactory.getCurrentSession();
+        } catch (HibernateException e) {}
+
         boolean startNewTransaction = true;
```

Comment:
You can remove var startNewTransaction and use (transaction != null) like you have done on line 60